### PR TITLE
add overlinking_ignore_patterns build parameter

### PR DESCRIPTION
**Pull request by msarahan**
### Description

Adds a new `build.overlinking_ignore_patterns` key to ignore file patterns for the overlinking and overdepending checks. This is sometimes useful to speed up builds that have many files (large repackage jobs) or builds where you know only a small fraction of the files should be checked.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

**Comment by msarahan** [comment]
CI failure is a CRAN network hiccup. Ready for review here.

File: tests/test-recipes/metadata/_overlinking_detection_ignore_patterns/build_scripts/with_bzip2.bat [hunk 1]
```diff
@@ -0,0 +1,5 @@
+@echo on
+echo #include ^<bzlib.h^>>main.c
```
Comment (kenodegard) [inline tests/test-recipes/metadata/_overlinking_detection_ignore_patterns/build_scripts/with_bzip2.bat:2]: Ugh, windows cmd quoting 🤦🏼‍♂️

**Review by kenodegard — CHANGES_REQUESTED** [review]
Minor edit, otherwise it looks good to me.

@chenghlee can you also take a look?

File: tests/test-recipes/metadata/_overlinking_detection_ignore_patterns/build_scripts/with_bzip2.bat (hunk 1 above)
Comment (msarahan) [reply tests/test-recipes/metadata/_overlinking_detection_ignore_patterns/build_scripts/with_bzip2.bat:2]: yeah. To be clear, that is just a straight copy from the existing _overlinking test. It's just adapted here to ignore the error/warning that normally shows up.

Comment (kenodegard) [reply tests/test-recipes/metadata/_overlinking_detection_ignore_patterns/build_scripts/with_bzip2.bat:2]: gotcha, good to know

**Review by kenodegard — APPROVED** [review]

**Pull request merged**